DTGB-777: Fixed filter layout overflow issue.

delrueba opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

![image](https://user-images.githubusercontent.com/31842807/67873439-c0ea5f80-fb33-11e9-854d-6f06c663314b.png)
![image](https://user-images.githubusercontent.com/31842807/67873465-c9429a80-fb33-11e9-8600-84106d16d348.png)


## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have updated the documentation accordingly.
- [x] I have updated the styleguide CHANGELOG accordingly.

delrueba committed:
DTGB-777: Fixed filter layout overflow issue.

delrueba committed:
DTGB-777: Updated changelog.

hstandaert commented on components/61-layouts/filter-layout/_filter-layout.scss:
```diff
@@ -12,6 +12,10 @@
     }
+
+    &.modal {
+      max-height: unset;
```
`unset` is not supported for IE.... https://developer.mozilla.org/en-US/docs/Web/CSS/unset

```suggestion
      max-height: none;
```

delrueba committed:
DTGB-777: Fixed keyboard bug on checkboxes with filter.
Eventlistener was not removed on submit, prevent enterkey has become obsolete in new modal.

delrueba committed:
Update components/61-layouts/filter-layout/_filter-layout.scss

Co-Authored-By: Helena Standaert <helena.standaert@gmail.com>
